### PR TITLE
netmap: sort buckets before applying HRW

### PR DESCRIPTION
**Pull request by fyrchik**
If weights for some buckets are equal, result depends on the
initial positions of buckets in slice.

Props to @ZhangTao1596.

Signed-off-by: Evgenii Stratonikov <evgeniy@nspcc.ru>

### CODE REVIEW

**Comment by fyrchik** [comment]
Sorting by HRW value (instead of index) makes `GetContainerNodes` a bit faster.
```
name             old time/op    new time/op    delta
PolicyHRWType-8    53.7µs ± 4%    51.9µs ± 4%  -3.49%  (p=0.000 n=19+19)

name             old alloc/op   new alloc/op   delta
PolicyHRWType-8    21.6kB ± 0%    20.6kB ± 0%  -4.70%  (p=0.000 n=19+20)

name             old allocs/op  new allocs/op  delta
PolicyHRWType-8       421 ± 0%       407 ± 0%  -3.33%  (p=0.000 n=20+20)
```

**Comment by alexvanin** [comment]
Sorting by value is the way to go. Should be there from the start.

File: netmap/selector_test.go [hunk 1]
```diff
@@ -296,6 +473,7 @@ func TestPlacementPolicy_ProcessSelectors(t *testing.T) {
 }
 
 func TestPlacementPolicy_ProcessSelectorsHRW(t *testing.T) {
+	t.Skip()
```
Comment (carpawell) [inline netmap/selector_test.go:476]: what is this for?

Comment (fyrchik) [reply netmap/selector_test.go:476]: I added it while making some changes, will remove.

**Comment by fyrchik** [comment]
I figured out, there was a mistake in benchmarks. HRW library silently returns when elements of slice provided to `SortSliceByWeightValue` do not implement `Hasher` interface. After I had added `Nodes.Hash` method, I got following results (`netmapSize = 1000`):
```
BenchmarkHRWSort/sort_by_index,_no_weight
BenchmarkHRWSort/sort_by_index,_no_weight-8         	    8715	    133716 ns/op
BenchmarkHRWSort/sort_by_value,_no_weight
BenchmarkHRWSort/sort_by_value,_no_weight-8         	    6654	    186734 ns/op
BenchmarkHRWSort/only_sort_by_index
BenchmarkHRWSort/only_sort_by_index-8               	    5784	    211262 ns/op
BenchmarkHRWSort/sort_by_value
BenchmarkHRWSort/sort_by_value-8                    	    4398	    262967 ns/op
BenchmarkHRWSort/sort_by_ID,_then_by_index_(deterministic)
BenchmarkHRWSort/sort_by_ID,_then_by_index_(deterministic)-8         	    4540	    262614 ns/op
```
So it turns out sorting by value is a bit slower than sorting by index. But we really need to compare implementations with _deterministic_ results (last 2 lines). And here the difference is not so significant. If we set `netmapSize = 10_000` in benchmarks, `sort_by_value` starts to be faster.

**Review by alexvanin — APPROVED** [review]
Took me a while to remember some placement function details, but it seems legit.

**Review by carpawell — APPROVED** [review]

**Pull request merged**